virtwho/configure.py

```diff
@@ -1,4 +1,144 @@
+        """
+        self.cfg.update(self.section, option, value)
+
+    def delete(self, option):
+        """Delete an option
+        :param option: Option to remove.
+        """
+        self.cfg.delete(self.section, option)
+
+    def destroy(self):
+        """Remove both the local and remote files"""
+        os.remove(self.local_file)
+        self.remote_ssh.remove_file(self.remote_file)
+
+
+class VirtwhoGlobalConfig:
+    """Able to manage /etc/virt-who.conf file when call this class"""
+
+    def __init__(self, mode=None):
+        """virt-who.conf file is backed up to /temp directory of the
+        project root.
+        :param mode: Hypervisor mode. When mode is local, will manage
+            the local libvirt host, othervise will manage the host for
+            all other remote modes.
+        """
+        self.mode = mode
+        self.remote_ssh = virtwho_ssh_connect(self.mode)
+        self.local_file = os.path.join(TEMP_DIR, 'virt-who.conf')
+        self.remote_file = '/etc/virt-who.conf'
+        self.save_file = os.path.join(TEMP_DIR, 'virt-who.conf.save')
+        if not os.path.exists(self.save_file):
+            self.remote_ssh.get_file(self.remote_file, self.save_file)
+        self.cfg = Configure(self.local_file, self.remote_ssh, self.remote_file)
+
+    def update(self, section, option, value):
+        """Add section, add option or update option
+        :param section: Section will be added if not exist.
+        :param option: Option will be added if not exist.
+        :param value: Value to update for the option.
+        """
+        self.cfg.update(section, option, value)
+
+    def delete(self, section, option=None):
+        """Remove a section or option
+        :param section: Section will be removed when no option provided.
+        :param option: Option to remove.
+        """
+        self.cfg.delete(section, option)
+
+
+def virtwho_ssh_connect(mode=None):
+    """Define the ssh connection of virt-who host, get data from
+    virtwho.ini file.
+    :param mode: The test hypervisor mode.
+    """
+    virtwho = config.virtwho
+    if mode == 'local':
+        virtwho = config.local
+        port = 22
+    else:
+        port = virtwho.port or 22
+    host = virtwho.host
+    username = virtwho.username
+    password = virtwho.password
+    return SSHConnect(host=host, user=username, pwd=password, port=port)
+
+
+def get_register_handler(register_type):
+    """Navigate to register type section in virtwho.ini.
+    :param register_type: rhsm or satellite. rhsm as default.
+    :return: register section
+    """
+    register = config.rhsm
+    if register_type == 'satellite':
+        register = config.satellite
+    return register
+
+
+def get_hypervisor_handler(mode):
+    """Navigate to hypervisor mode section in virtwho.ini.
+    :param mode: The hypervisor mode. esx as default
+    :return: hypervisor section
+    """
+    hypervisor = config.vcenter
+    if mode == 'xen':
```

Comment:
There is a good way to address this function, but it is up to you :)
```
if mode in ['vcenter', 'xen', 'hyperv', 'rhevm', 'libvirt', 'kubevirt']:
    return getattr(config, mode)
```

Reply:
Yes, it's a better way, and updated to use getattr().